component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -336,6 +327,135 @@ protected void processAuthenticationResponse(HttpServletRequest request, HttpSer
         }
     }
 
+    /**
+     * Process valid user token.
+     *
+     * @param context           AuthenticationContext.
+     * @param authenticatedUser AuthenticatedUser.
+     */
+    private void processValidUserToken(AuthenticationContext context, AuthenticatedUser authenticatedUser) {
+
+        context.setProperty(EmailOTPAuthenticatorConstants.OTP_TOKEN, EmailOTPAuthenticatorConstants.EMPTY_STRUNG);
+        context.setProperty(EmailOTPAuthenticatorConstants.EMAILOTP_ACCESS_TOKEN,
+                EmailOTPAuthenticatorConstants.EMPTY_STRUNG);
+        context.setSubject(authenticatedUser);
+        String emailFromProfile = context.getProperty(EmailOTPAuthenticatorConstants.RECEIVER_EMAIL).toString();
+        context.setSubject(
+                AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(emailFromProfile));
+    }
+
+    /**
+     * Checks the backup codes for email otp.
+     *
+     * @param context           AuthenticationContext.
+     * @param userToken         EmailOTP token.
+     * @param authenticatedUser AuthenticatedUser
+     * @throws AuthenticationFailedException
+     */
+    private void checkWithBackUpCodes(AuthenticationContext context, String userToken,
+                                      AuthenticatedUser authenticatedUser) throws AuthenticationFailedException {
+
+        String[] savedOTPs = null;
+        String username = authenticatedUser.getUserName();
+        String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
+        UserRealm userRealm = getUserRealm(username);
+        try {
+            if (userRealm != null) {
+                UserStoreManager userStoreManager = userRealm.getUserStoreManager();
```

Comment:
does this work correctly for a secondary userstore user? You may need to use [1] approach for retrieving claims of a secondary userstore user.
https://github.com/wso2/carbon-identity-framework/blob/master/components/user-mgt/org.wso2.carbon.identity.user.profile/src/main/java/org/wso2/carbon/identity/user/profile/mgt/UserProfileAdmin.java#L425

Reply:
Yes. It is working. It is internally handled in ABstractUserstore manager https://github.com/wso2-support/carbon4-kernel/blob/support-4.4.40/core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java#L762